### ISSUE TITLE
cmd/compile: incorrect typecheck error for multiple assignment

### ISSUE THREAD

**Issue by rsc** [golang/go#43348]
```
% cat /tmp/x.go
package p

func f() string

func _() {
	x := f(), 1, 2
}
% go tool compile /tmp/x.go
/tmp/x.go:6:4: assignment mismatch: 1 variables but f returns 3 values
% 
```

Narrator: f does not return 3 values.

Been this way back to Go 1.12 at least (I don't have earlier versions on my Mac laptop, and honestly I'm aamzed Go 1.12 still runs).

/cc @griesemer @mdempsky

**Comment by ianlancetaylor** [comment]
Go 1.11 is better:

```
foo.go:6:4: assignment mismatch: 1 variables but 3 values
```

gccgo says

```
foo.go:6:9: error: wrong number of initializations
    6 |         x := f(), 1, 2
      |         ^
```

**Comment by griesemer** [comment]
The compiler with types2 reports:
```
$ go tool compile -G main.go
main.go:6:2: x declared but not used
main.go:6:7: cannot initialize 1 variables with 3 values
```
I think this is about as good as the old error message.
(I agree that the error about `x` not being used doesn't need to be there. That's a separate issue.)

**Comment by mdempsky** [comment]
This is fixed on dev.regabi:

```
$ go tool compile /tmp/x.go
/tmp/x.go:6:4: assignment mismatch: 1 variable but 3 values
```

**Comment by robpike** [comment]
@rsc I can report it goes at least go Go 1.4.

**Comment by tv42** [comment]
This seems closed in error, if it's not fixed in master? Or has dev.regabi become a de facto master?

I just checked and 44361140c02556a0a71bc52299149bb8de26024b still has this bug.

**Comment by griesemer** [comment]
Per [the comment](https://github.com/golang/go/issues/43348#issuecomment-762064598
) above, this was fixed in `dev.regabi`. This will be fixed for Go 1.17.
